### PR TITLE
add throttling, token counters to operation

### PR DESCRIPTION
**Pull request by ryanfkeepers**
Adds context embedding to pkg/count, and embeds
the backup, restore, and export operation ctx with the operations count.bus.  The bus is extracted from the ctx in api middleware to track throttling response counts, as well as token consumption counts.  Counts are globally tracked, not on a per-time or per resource basis.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/pkg/count/keys.go [hunk 1]
```diff
@@ -3,10 +3,12 @@ package count
 type key string
 
 const (
+	APICallTokensConsumed key = "api-call-tokens-consumed"
```
Comment (pandeyabs) [inline src/pkg/count/keys.go:6]: Could you add a small description for each of these?

File: src/pkg/count/count.go [hunk 2]
```diff
@@ -106,3 +108,33 @@ func (b *Bus) TotalValues() map[string]int64 {
 
 	return b.Values()
 }
+
+// ---------------------------------------------------------------------------
+// context embedding
+// ---------------------------------------------------------------------------
+
+type countKey string
```
Comment (pandeyabs) [inline src/pkg/count/count.go:116]: `countBusKey` instead of `countKey`?

**Review by pandeyabs — APPROVED** [review]
LGTM. Great pattern!

File: src/pkg/count/count.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/pkg/count/count.go:116]: Eh, it's unexported.  Could just be named `key`.  But I don't think it's critical either way.

File: src/pkg/count/keys.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/pkg/count/keys.go:6]: Sure, why not.

**Pull request merged**